src/main/java/com/gravity9/mongocse/DefaultMongoClientProvider.java

```diff
@@ -0,0 +1,52 @@
+package com.gravity9.mongocse;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.PojoCodecProvider;
+
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+class DefaultMongoClientProvider implements MongoClientProvider {
+
+	private final String mongoConnectionUri;
+
+	private MongoClient client;
+
+    DefaultMongoClientProvider(String mongoConnectionUri) {
+        this.mongoConnectionUri = mongoConnectionUri;
+    }
+
+	@Override
+    public MongoClient getClient() {
+		if (client == null) {
```

Comment:
this part is not thread-safe, maybe LazyHolder?
* https://stackoverflow.com/a/8297830

Reply:
I preferred synchronization on provider instance, to ensure both cration and destruction is synchronized as well.